core/src/cluster_slot_state_verifier.rs

```diff
@@ -33,147 +45,185 @@ impl SlotStateUpdate {
     fn to_handler(&self) -> SlotStateHandler {
         match self {
             SlotStateUpdate::Dead => on_dead_slot,
-            SlotStateUpdate::Frozen => on_frozen_slot,
-            SlotStateUpdate::DuplicateConfirmed => on_cluster_update,
-            SlotStateUpdate::Duplicate => on_cluster_update,
+            SlotStateUpdate::BankFrozen => on_frozen_slot,
+            SlotStateUpdate::DuplicateConfirmed => on_duplicate_confirmed,
+            SlotStateUpdate::Duplicate => on_duplicate,
         }
     }
 }
 
-fn on_dead_slot(
+fn check_duplicate_confirmed_hash_against_frozen_hash(
```

Comment:
This function does too much in my opinion. As a start, how about this is only called when we _do_ have a cluster duplicate confirmed hash?

Reply:
Yeah, agreed, updated!